apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowFilters.cs

```diff
@@ -0,0 +1,380 @@
+            new RowFilter { ColumnRangeFilter = GaxPreconditions.CheckNotNull(range, nameof(range)) };
+
+        // TODO: Add additional ColumnRange... methods to take start/end values?
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which applies one of two
+        /// possible RowFilters to the data based on the output of a predicate
+        /// RowFilter.
+        /// </summary>
+        /// <param name="predicateFilter">
+        /// If <paramref name="predicateFilter"/> outputs any cells, then
+        /// <paramref name="trueFilter"/> will be evaluated on the input row.
+        /// Otherwise, <paramref name="falseFilter"/> will be evaluated.
+        /// Must not be null.
+        /// </param>
+        /// <param name="trueFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// returns any results. If null, no results will be returned in the true case.
+        /// </param>
+        /// <param name="falseFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// does not return any results. If null, no results will be returned in the
+        /// false case.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter Condition(
+            RowFilter predicateFilter, RowFilter trueFilter, RowFilter falseFilter) =>
+            new RowFilter
+            {
+                Condition = new RowFilter.Types.Condition
+                {
+                    // TODO: Verify this on server. Might be allowed for (row contains any cells).
+                    PredicateFilter = GaxPreconditions.CheckNotNull(predicateFilter, nameof(predicateFilter)),
+                    TrueFilter = trueFilter,
+                    FalseFilter = falseFilter
+                }
+            };
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only cells
+        /// from columns whose families satisfy the given RE2 regex.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// For technical reasons, the regex must not contain the `:` character, even
+        /// if it is not being used as a literal. Note that, since column families
+        /// cannot contain the new line character `\n`, it is sufficient to use `.`
+        /// as a full wildcard when matching column family names.
+        /// </para>
+        /// </remarks>
+        /// <param name="regex">
+        /// The RE2 regex used to match column families. Must not be null.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter FamilyNameRegex(string regex) =>
+            new RowFilter { FamilyNameRegexFilter = GaxPreconditions.CheckNotNull(regex, nameof(regex)) };
+
+        // TODO: Copy docs from Interleave.Filters, but only once the formatting issues are resolved.
```

Comment:
Definitely - I've no idea what this means at the moment compared with Chain :)

Reply:
You can think of Chain like and AND boolean operator, but the filters are ordered. Interleave can be thought of like OR, but if multiple filters match a cell, it will appear in the output once for each of those matching filters. Chain will only "output" a cell if it flows through each filter in the chain, in order. Interleave works in parallel and outputs each cell that passes each filter. There is more info about interleave here: https://cloud.google.com/bigtable/docs/reference/data/rpc/google.bigtable.v2#google.bigtable.v2.RowFilter.Interleave, but it just doesn't format correctly, so I need to figure out the best way to include it.